sql: TestDistBackfill failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/99d39b356eab0ad449f304df8585323da0b30d6c

Parameters:
```
TAGS=
GOFLAGS=-race
```

Stress build found a failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=466506&tab=buildLog

```
W180104 09:00:34.420424 7017 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180104 09:00:34.429020 7017 server/config.go:516  [n?] 1 storage engine initialized
I180104 09:00:34.429163 7017 server/config.go:519  [n?] RocksDB cache size: 128 MiB
I180104 09:00:34.429234 7017 server/config.go:519  [n?] store 0: in-memory, size 0 B
I180104 09:00:34.517379 7017 server/node.go:361  [n?] **** cluster c57370e7-77ce-4bd7-8201-62bfad5d818d has been created
I180104 09:00:34.518233 7017 server/server.go:934  [n?] **** add additional nodes by specifying --join=127.0.0.1:46111
I180104 09:00:34.646638 7017 server/node.go:486  [n1] initialized store [n1,s1]: disk (capacity=512 MiB, available=512 MiB, used=0 B, logicalBytes=3.2 KiB), ranges=1, leases=1, writes=0.00, bytesPerReplica={p10=3322.00 p25=3322.00 p50=3322.00 p75=3322.00 p90=3322.00}, writesPerReplica={p10=0.00 p25=0.00 p50=0.00 p75=0.00 p90=0.00}
I180104 09:00:34.647242 7017 server/node.go:339  [n1] node ID 1 initialized
I180104 09:00:34.647947 7017 gossip/gossip.go:332  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:46111" > attrs:<> locality:<> ServerVersion:<major_val:1 minor_val:1 patch:0 unstable:8 > 
I180104 09:00:34.650425 7017 storage/stores.go:331  [n1] read 0 node addresses from persistent storage
I180104 09:00:34.652015 7017 server/node.go:627  [n1] connecting to gossip network to verify cluster ID...
I180104 09:00:34.652580 7017 server/node.go:652  [n1] node connected via gossip and verified as part of cluster "c57370e7-77ce-4bd7-8201-62bfad5d818d"
I180104 09:00:34.677202 7017 server/node.go:428  [n1] node=1: started with [<no-attributes>=<in-mem>] engine(s) and attributes []
I180104 09:00:34.688957 7017 sql/distsql_physical_planner.go:122  [n1] creating DistSQLPlanner with address {tcp 127.0.0.1:46111}
I180104 09:00:34.732970 7017 server/server.go:1161  [n1] starting https server at 127.0.0.1:37301
I180104 09:00:34.733151 7017 server/server.go:1162  [n1] starting grpc/postgres server at 127.0.0.1:46111
I180104 09:00:34.733203 7017 server/server.go:1163  [n1] advertising CockroachDB node at 127.0.0.1:46111
W180104 09:00:34.763753 7017 sql/jobs/registry.go:222  [n1] unable to get node liveness: node not in the liveness table
I180104 09:00:34.991736 7017 sql/event_log.go:115  [n1] Event: "alter_table", target: 12, info: {TableName:eventlog Statement:ALTER TABLE system.eventlog ALTER COLUMN "uniqueID" SET DEFAULT uuid_v4() User:node MutationID:0 CascadeDroppedViews:[]}
I180104 09:00:35.042925 7017 sql/lease.go:348  [n1] publish: descID=12 (eventlog) version=2 mtime=2018-01-04 09:00:35.041412369 +0000 UTC
I180104 09:00:35.343577 7017 sql/event_log.go:115  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:diagnostics.reporting.enabled Value:true User:node}
I180104 09:00:35.491891 7017 sql/event_log.go:115  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:version Value:$1 User:node}
I180104 09:00:35.620285 7017 sql/event_log.go:115  [n1] Event: "set_cluster_setting", target: 0, info: {SettingName:trace.debug.enable Value:false User:node}
I180104 09:00:35.884249 7017 sql/event_log.go:115  [n1] Event: "alter_table", target: 4, info: {TableName:users Statement:ALTER TABLE system.users ADD COLUMN IF NOT EXISTS "isRole" BOOL NOT NULL DEFAULT false User:node MutationID:1 CascadeDroppedViews:[]}
I180104 09:00:35.927114 7017 sql/lease.go:348  [n1] publish: descID=4 (users) version=2 mtime=2018-01-04 09:00:35.9263864 +0000 UTC
I180104 09:00:35.974336 7017 sql/lease.go:348  [n1] publish: descID=4 (users) version=3 mtime=2018-01-04 09:00:35.973584765 +0000 UTC
I180104 09:00:36.022414 7017 sql/backfill.go:132  [n1] Running backfill for "users", v=3, m=1
I180104 09:00:36.250089 7017 sql/lease.go:348  [n1] publish: descID=4 (users) version=4 mtime=2018-01-04 09:00:36.24886579 +0000 UTC
I180104 09:00:36.410875 7017 sql/event_log.go:115  [n1] Event: "finish_schema_change", target: 4, info: {MutationID:1}
I180104 09:00:36.438373 7017 sql/lease.go:274  publish (count leases): descID=4 name=users version=3 count=1
I180104 09:00:36.671480 7017 server/server.go:1232  [n1] done ensuring all necessary migrations have run
I180104 09:00:36.671735 7017 server/server.go:1235  [n1] serving sql connections
I180104 09:00:36.729972 7179 sql/event_log.go:115  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:46111} Attrs: Locality: ServerVersion:1.1-8} ClusterID:c57370e7-77ce-4bd7-8201-62bfad5d818d StartedAt:1515056434665954040 LastUp:1515056434665954040}
W180104 09:00:36.765836 7017 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180104 09:00:36.774737 7017 server/config.go:516  [n?] 1 storage engine initialized
I180104 09:00:36.774852 7017 server/config.go:519  [n?] RocksDB cache size: 128 MiB
I180104 09:00:36.774924 7017 server/config.go:519  [n?] store 0: in-memory, size 0 B
W180104 09:00:36.775246 7017 gossip/gossip.go:1292  [n?] no incoming or outgoing connections
I180104 09:00:36.776902 7017 server/server.go:936  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I180104 09:00:36.872697 6928 gossip/client.go:129  [n?] started gossip client to 127.0.0.1:46111
I180104 09:00:36.875303 7436 gossip/server.go:219  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:44117}
I180104 09:00:36.903771 7017 storage/stores.go:331  [n?] read 0 node addresses from persistent storage
I180104 09:00:36.905297 7017 storage/stores.go:350  [n?] wrote 1 node addresses to persistent storage
I180104 09:00:36.905381 7017 server/node.go:627  [n?] connecting to gossip network to verify cluster ID...
I180104 09:00:36.905558 7017 server/node.go:652  [n?] node connected via gossip and verified as part of cluster "c57370e7-77ce-4bd7-8201-62bfad5d818d"
I180104 09:00:36.909383 6929 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:36.915809 7509 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:36.921850 7017 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:36.929809 7017 server/node.go:332  [n?] new node allocated ID 2
I180104 09:00:36.930442 7017 gossip/gossip.go:332  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:44117" > attrs:<> locality:<> ServerVersion:<major_val:1 minor_val:1 patch:0 unstable:8 > 
I180104 09:00:36.931229 7017 server/node.go:414  [n2] node=2: asynchronously bootstrapping engine(s) [<no-attributes>=<in-mem>]
I180104 09:00:36.931916 7017 server/node.go:428  [n2] node=2: started with [] engine(s) and attributes []
I180104 09:00:36.932986 7017 sql/distsql_physical_planner.go:122  [n2] creating DistSQLPlanner with address {tcp 127.0.0.1:44117}
I180104 09:00:36.945217 7181 storage/stores.go:350  [n1] wrote 1 node addresses to persistent storage
I180104 09:00:37.012752 7017 server/server.go:1161  [n2] starting https server at 127.0.0.1:46249
I180104 09:00:37.014582 7017 server/server.go:1162  [n2] starting grpc/postgres server at 127.0.0.1:44117
I180104 09:00:37.014682 7017 server/server.go:1163  [n2] advertising CockroachDB node at 127.0.0.1:44117
I180104 09:00:37.021713 7017 server/server.go:1232  [n2] done ensuring all necessary migrations have run
I180104 09:00:37.021874 7017 server/server.go:1235  [n2] serving sql connections
I180104 09:00:37.086823 6908 server/node.go:608  [n2] bootstrapped store [n2,s2]
I180104 09:00:37.192074 6638 sql/event_log.go:115  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:44117} Attrs: Locality: ServerVersion:1.1-8} ClusterID:c57370e7-77ce-4bd7-8201-62bfad5d818d StartedAt:1515056436931498117 LastUp:1515056436931498117}
W180104 09:00:37.210021 7017 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180104 09:00:37.262257 7017 server/config.go:516  [n?] 1 storage engine initialized
I180104 09:00:37.266415 7017 server/config.go:519  [n?] RocksDB cache size: 128 MiB
I180104 09:00:37.266575 7017 server/config.go:519  [n?] store 0: in-memory, size 0 B
W180104 09:00:37.266929 7017 gossip/gossip.go:1292  [n?] no incoming or outgoing connections
I180104 09:00:37.267638 7017 server/server.go:936  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I180104 09:00:37.427797 7464 gossip/client.go:129  [n?] started gossip client to 127.0.0.1:46111
I180104 09:00:37.431949 7748 gossip/server.go:219  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:39187}
I180104 09:00:37.456852 7017 storage/stores.go:331  [n?] read 0 node addresses from persistent storage
I180104 09:00:37.457323 7017 storage/stores.go:350  [n?] wrote 2 node addresses to persistent storage
I180104 09:00:37.457453 7017 server/node.go:627  [n?] connecting to gossip network to verify cluster ID...
I180104 09:00:37.457614 7017 server/node.go:652  [n?] node connected via gossip and verified as part of cluster "c57370e7-77ce-4bd7-8201-62bfad5d818d"
I180104 09:00:37.462959 7766 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:37.468902 7765 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:37.474237 7017 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:37.495800 7017 server/node.go:332  [n?] new node allocated ID 3
I180104 09:00:37.496382 7017 gossip/gossip.go:332  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:39187" > attrs:<> locality:<> ServerVersion:<major_val:1 minor_val:1 patch:0 unstable:8 > 
I180104 09:00:37.497213 7017 server/node.go:414  [n3] node=3: asynchronously bootstrapping engine(s) [<no-attributes>=<in-mem>]
I180104 09:00:37.497928 7017 server/node.go:428  [n3] node=3: started with [] engine(s) and attributes []
I180104 09:00:37.498870 7017 sql/distsql_physical_planner.go:122  [n3] creating DistSQLPlanner with address {tcp 127.0.0.1:39187}
I180104 09:00:37.527236 7781 storage/stores.go:350  [n1] wrote 2 node addresses to persistent storage
I180104 09:00:37.531116 7195 storage/stores.go:350  [n2] wrote 2 node addresses to persistent storage
I180104 09:00:37.585022 7017 server/server.go:1161  [n3] starting https server at 127.0.0.1:35745
I180104 09:00:37.585252 7017 server/server.go:1162  [n3] starting grpc/postgres server at 127.0.0.1:39187
I180104 09:00:37.585342 7017 server/server.go:1163  [n3] advertising CockroachDB node at 127.0.0.1:39187
I180104 09:00:37.593424 7017 server/server.go:1232  [n3] done ensuring all necessary migrations have run
I180104 09:00:37.593611 7017 server/server.go:1235  [n3] serving sql connections
I180104 09:00:37.602229 7472 server/node.go:608  [n3] bootstrapped store [n3,s3]
I180104 09:00:37.686057 7960 sql/event_log.go:115  [n3] Event: "node_join", target: 3, info: {Descriptor:{NodeID:3 Address:{NetworkField:tcp AddressField:127.0.0.1:39187} Attrs: Locality: ServerVersion:1.1-8} ClusterID:c57370e7-77ce-4bd7-8201-62bfad5d818d StartedAt:1515056437497530662 LastUp:1515056437497530662}
W180104 09:00:37.782559 7017 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180104 09:00:37.809427 7017 server/config.go:516  [n?] 1 storage engine initialized
I180104 09:00:37.809585 7017 server/config.go:519  [n?] RocksDB cache size: 128 MiB
I180104 09:00:37.809628 7017 server/config.go:519  [n?] store 0: in-memory, size 0 B
W180104 09:00:37.810006 7017 gossip/gossip.go:1292  [n?] no incoming or outgoing connections
I180104 09:00:37.810814 7017 server/server.go:936  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I180104 09:00:38.012390 7752 gossip/client.go:129  [n?] started gossip client to 127.0.0.1:46111
I180104 09:00:38.015783 7769 gossip/server.go:219  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:43697}
I180104 09:00:38.046717 7017 storage/stores.go:331  [n?] read 0 node addresses from persistent storage
I180104 09:00:38.047097 7017 storage/stores.go:350  [n?] wrote 1 node addresses to persistent storage
I180104 09:00:38.047250 7017 server/node.go:627  [n?] connecting to gossip network to verify cluster ID...
I180104 09:00:38.047425 7017 server/node.go:652  [n?] node connected via gossip and verified as part of cluster "c57370e7-77ce-4bd7-8201-62bfad5d818d"
I180104 09:00:38.056897 7405 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:38.063179 7404 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:38.065135 8050 storage/stores.go:350  [n?] wrote 2 node addresses to persistent storage
I180104 09:00:38.072470 8050 storage/stores.go:350  [n?] wrote 3 node addresses to persistent storage
I180104 09:00:38.085411 7017 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:38.102644 7017 server/node.go:332  [n?] new node allocated ID 4
I180104 09:00:38.103262 7017 gossip/gossip.go:332  [n4] NodeDescriptor set to node_id:4 address:<network_field:"tcp" address_field:"127.0.0.1:43697" > attrs:<> locality:<> ServerVersion:<major_val:1 minor_val:1 patch:0 unstable:8 > 
I180104 09:00:38.104917 7017 server/node.go:414  [n4] node=4: asynchronously bootstrapping engine(s) [<no-attributes>=<in-mem>]
I180104 09:00:38.105665 7017 server/node.go:428  [n4] node=4: started with [] engine(s) and attributes []
I180104 09:00:38.106885 7017 sql/distsql_physical_planner.go:122  [n4] creating DistSQLPlanner with address {tcp 127.0.0.1:43697}
I180104 09:00:38.107823 7407 storage/stores.go:350  [n1] wrote 3 node addresses to persistent storage
I180104 09:00:38.111370 8037 storage/stores.go:350  [n3] wrote 3 node addresses to persistent storage
I180104 09:00:38.119831 7408 storage/stores.go:350  [n2] wrote 3 node addresses to persistent storage
I180104 09:00:38.212672 7017 server/server.go:1161  [n4] starting https server at 127.0.0.1:42217
I180104 09:00:38.212872 7017 server/server.go:1162  [n4] starting grpc/postgres server at 127.0.0.1:43697
I180104 09:00:38.212928 7017 server/server.go:1163  [n4] advertising CockroachDB node at 127.0.0.1:43697
I180104 09:00:38.224090 7501 server/node.go:608  [n4] bootstrapped store [n4,s4]
I180104 09:00:38.311783 7017 server/server.go:1232  [n4] done ensuring all necessary migrations have run
I180104 09:00:38.312030 7017 server/server.go:1235  [n4] serving sql connections
I180104 09:00:38.559556 7736 sql/event_log.go:115  [n4] Event: "node_join", target: 4, info: {Descriptor:{NodeID:4 Address:{NetworkField:tcp AddressField:127.0.0.1:43697} Attrs: Locality: ServerVersion:1.1-8} ClusterID:c57370e7-77ce-4bd7-8201-62bfad5d818d StartedAt:1515056438105227451 LastUp:1515056438105227451}
W180104 09:00:38.650690 7017 server/status/runtime.go:109  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I180104 09:00:38.677147 7017 server/config.go:516  [n?] 1 storage engine initialized
I180104 09:00:38.677281 7017 server/config.go:519  [n?] RocksDB cache size: 128 MiB
I180104 09:00:38.677357 7017 server/config.go:519  [n?] store 0: in-memory, size 0 B
W180104 09:00:38.677700 7017 gossip/gossip.go:1292  [n?] no incoming or outgoing connections
I180104 09:00:38.678444 7017 server/server.go:936  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I180104 09:00:38.782042 8264 gossip/client.go:129  [n?] started gossip client to 127.0.0.1:46111
I180104 09:00:38.785466 6701 gossip/server.go:219  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:46439}
I180104 09:00:38.814400 7017 storage/stores.go:331  [n?] read 0 node addresses from persistent storage
I180104 09:00:38.814915 7017 storage/stores.go:350  [n?] wrote 4 node addresses to persistent storage
I180104 09:00:38.815096 7017 server/node.go:627  [n?] connecting to gossip network to verify cluster ID...
I180104 09:00:38.815406 7017 server/node.go:652  [n?] node connected via gossip and verified as part of cluster "c57370e7-77ce-4bd7-8201-62bfad5d818d"
I180104 09:00:38.818432 8338 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:38.824256 8241 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:38.830295 7017 kv/dist_sender.go:354  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I180104 09:00:38.841380 7017 server/node.go:332  [n?] new node allocated ID 5
I180104 09:00:38.841973 7017 gossip/gossip.go:332  [n5] NodeDescriptor set to node_id:5 address:<network_field:"tcp" address_field:"127.0.0.1:46439" > attrs:<> locality:<> ServerVersion:<major_val:1 minor_val:1 patch:0 unstable:8 > 
I180104 09:00:38.843255 7017 server/node.go:414  [n5] node=5: asynchronously bootstrapping engine(s) [<no-attributes>=<in-mem>]
I180104 09:00:38.845168 6701 gossip/server.go:269  [n1] refusing gossip from node 5 (max 3 conns); forwarding to 3 ({tcp 127.0.0.1:39187})
I180104 09:00:38.846993 8264 gossip/client.go:134  [n5] closing client to node 1 (127.0.0.1:46111): received forward from node 1 to 3 (127.0.0.1:39187)
I180104 09:00:38.853907 8281 gossip/gossip.go:1306  [n5] node has connected to cluster via gossip
I180104 09:00:38.854623 8281 storage/stores.go:350  [n5] wrote 4 node addresses to persistent storage
I180104 09:00:38.857804 8312 storage/stores.go:350  [n1] wrote 4 node addresses to persistent storage
I180104 09:00:38.860686 7724 storage/stores.go:350  [n4] wrote 4 node addresses to persistent storage
I180104 09:00:38.864371 7725 storage/stores.go:350  [n2] wrote 4 node addresses to persistent storage
I180104 09:00:38.867831 7726 storage/stores.go:350  [n3] wrote 4 node addresses to persistent storage
I180104 09:00:38.882667 7017 server/node.go:428  [n5] node=5: started with [] engine(s) and attributes []
I180104 09:00:38.883728 7017 sql/distsql_physical_planner.go:122  [n5] creating DistSQLPlanner with address {tcp 127.0.0.1:46439}
I180104 09:00:38.993090 7017 server/server.go:1161  [n5] starting https server at 127.0.0.1:40023
I180104 09:00:38.993266 7017 server/server.go:1162  [n5] starting grpc/postgres server at 127.0.0.1:46439
I180104 09:00:38.993315 7017 server/server.go:1163  [n5] advertising CockroachDB node at 127.0.0.1:46439
I180104 09:00:39.002798 7993 server/node.go:608  [n5] bootstrapped store [n5,s5]
I180104 09:00:39.118020 7017 server/server.go:1232  [n5] done ensuring all necessary migrations have run
I180104 09:00:39.118252 7017 server/server.go:1235  [n5] serving sql connections
I180104 09:00:39.298795 8105 sql/event_log.go:115  [n5] Event: "node_join", target: 5, info: {Descriptor:{NodeID:5 Address:{NetworkField:tcp AddressField:127.0.0.1:46439} Attrs: Locality: ServerVersion:1.1-8} ClusterID:c57370e7-77ce-4bd7-8201-62bfad5d818d StartedAt:1515056438868934408 LastUp:1515056438868934408}
I180104 09:00:39.317123 8041 gossip/client.go:129  [n5] started gossip client to 127.0.0.1:39187
I180104 09:00:39.810969 8591 sql/event_log.go:115  [client=127.0.0.1:60656,user=root,n1] Event: "create_database", target: 50, info: {DatabaseName:test Statement:CREATE DATABASE IF NOT EXISTS test User:root}
I180104 09:00:39.900883 8591 sql/event_log.go:115  [client=127.0.0.1:60656,user=root,n1] Event: "create_table", target: 51, info: {TableName:test.numtosquare Statement:CREATE TABLE test.numtosquare (x INT PRIMARY KEY, xsquared INT) User:root}
I180104 09:00:40.062920 8591 sql/event_log.go:115  [client=127.0.0.1:60656,user=root,n1] Event: "create_table", target: 52, info: {TableName:test.numtostr Statement:CREATE TABLE test.numtostr (y INT PRIMARY KEY, str STRING) User:root}


ERROR: signal: killed

context canceled
Makefile:804: recipe for target 'stress' failed
make: *** [stress] Error 1
```

--- a/pkg/sql/distsql_plan_backfill_test.go
+++ b/pkg/sql/distsql_plan_backfill_test.go
@@ -25,6 +25,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sqlbase"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
+	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 )
 
@@ -39,7 +40,11 @@ func TestDistBackfill(t *testing.T) {
 	//    squares
 	//  - a NumToStr table of size N^2 that maps integers to their string
 	//    representations. This table is split and distributed to all the nodes.
-	const n = 100
+	n := 100
+	if util.RaceEnabled {
+		// Race builds are a lot slower, so use a smaller number of rows.
+		n = 10
+	}
 	const numNodes = 5
 
 	tc := serverutils.StartTestCluster(t, numNodes,

